src/config.ts

```diff
@@ -407,136 +407,125 @@ const default_config = {
 
     /**
      * Characters to use in hint mode.
      *
      * They are used preferentially from left to right.
      */
-    hintchars: "hjklasdfgyuiopqwertnmzxcvb",
+    hintchars = "hjklasdfgyuiopqwertnmzxcvb"
 
     /**
      * The type of hinting to use. `vimperator` will allow you to filter links based on their names by typing non-hint chars. It is recommended that you use this in conjuction with the [[hintchars]] setting, which you should probably set to e.g, `5432167890`.
-     *
-     * Permitted values: `simple`, `vimperator`, or `vimperator-reflow`.
      */
-    hintfiltermode: "simple", // "simple", "vimperator", "vimperator-reflow"
+    hintfiltermode: "simple" | "vimperator" | "vimperator-reflow" = "simple"
 
     /**
      * Whether to optimise for the shortest possible names for each hint, or to use a simple numerical ordering. If set to `numeric`, overrides `hintchars` setting.
-     *
-     * Permitted values: `short` or `numeric`
      */
-    hintnames: "short",
+    hintnames: "short" | "numeric" = "short"
 
     /**
      * Whether to display the names for hints in uppercase.
      */
-    hintuppercase: "true",
+    hintuppercase: "true" | "false" = "true"
 
     /**
      * The delay in milliseconds in `vimperator` style hint modes after selecting a hint before you are returned to normal mode.
      *
      * The point of this is to prevent accidental execution of normal mode binds due to people typing more than is necessary to choose a hint.
      */
-    hintdelay: "300",
+    hintdelay = "300"
 
     /**
      * Controls whether the page can focus elements for you via js
      *
      * Best used in conjunction with browser.autofocus in `about:config`
-     *
-     * Permitted values: `true`, or `false`.
      */
-    allowautofocus: "true",
+    allowautofocus: "true" | "false" = "true"
 
     /**
      * Whether to use Tridactyl's (bad) smooth scrolling.
      */
-    smoothscroll: "false", // "false" | "true"
+    smoothscroll: "true" | "false" = "false"
 
     /**
      * How viscous you want smooth scrolling to feel.
      */
-    scrollduration: 100, // number
+    scrollduration = "100"
 
     /**
      * Where to open tabs opened with `tabopen` - to the right of the current tab, or at the end of the tabs.
-     *
-     * Permitted values: `next`, or `last`.
      */
-    tabopenpos: "next",
+    tabopenpos: "next" | "last" = "next"
 
     /**
      * Where to open tabs opened with hinting - as if it had been middle clicked, to the right of the current tab, or at the end of the tabs.
-     *
-     * Permitted values: `related, `next`, or `last`.
      */
-    relatedopenpos: "related",
-    ttsvoice: "default", // chosen from the listvoices list, or "default"
-    ttsvolume: 1, // 0 to 1
-    ttsrate: 1, // 0.1 to 10
-    ttspitch: 1, // 0 to 2
+    relatedopenpos: "related" | "next" | "last" = "related"
+    ttsvoice = "default" // chosen from the listvoices list or "default"
+    ttsvolume = 1 // 0 to 1
```

Comment:
`Unsupported setting type` which is true but we should probably make this a compiler error...

Reply:
i.e, on `set tts volume 1`

Reply:
If I ruled the world this would be as easy as `forAll T in class: T in any \ number` but it looks like you might be able to do something with conditional types - https://www.typescriptlang.org/docs/handbook/advanced-types.html#conditional-types

Reply:
Yup, I mentioned it in 3e9c212 . What's funny is that setting this setting has been broken since at list february 1st and nobody complained.

Now that I think about it, since we now have type information about the config, we could try to cast integer settings to Number. This would make accessing settings everywhere in Tridactyl slightly nicer as you wouldn't need to cast the string you get from `config.get()` anymore.